test/e2e/kubevirt.go

```diff
@@ -0,0 +1,628 @@
+
+			step = by(vm.Name, "Start httpServer")
+			httpServerCommand := fmt.Sprintf("podman run -d --tls-verify=false --privileged --net=host %s netexec --http-port %d", agnhostImage, httpServerPort)
+			output, err := kubevirt.RunCommand(kvcli, vmi, httpServerCommand, time.Minute)
+			Expect(err).ToNot(HaveOccurred(), step+": "+output)
+
+			step = by(vm.Name, "Expose httpServer as a service")
+			svc, err := fr.ClientSet.CoreV1().Services(namespace).Create(context.TODO(), composeService("httpserver", vm.Name, httpServerPort), metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred(), step)
+			endpoints, err := serviceEndpoints(svc)
+			Expect(err).ToNot(HaveOccurred(), step)
+
+			step = by(vm.Name, "Send first echo to populate connection pool")
+			Eventually(func() error { return sendEchoAndIgnoreReuse(endpoints) }).
+				WithPolling(1*time.Second).
+				WithTimeout(5*time.Second).
+				Should(Succeed(), step)
+
+			checkConnectivity(td, vm.Name, endpoints, "before live migration")
+
+			for i := uint(1); i <= td.numberOfMigrations; i++ {
+
+				by(vm.Name, fmt.Sprintf("Live migrate virtual machine #%d", i))
+				liveMigrateVirtualMachine(vm.Name, td.mode)
+
+				checkConnectivity(td, vm.Name, endpoints, fmt.Sprintf("after live migration #%d", i))
+			}
+
+		}
+	)
+	DescribeTable("when live migrated", func(td liveMigrationTestData) {
+		if td.mode == kubevirtv1.MigrationPostCopy && os.Getenv("GITHUB_ACTIONS") == "true" {
+			Skip("Post copy live migration not working at github actions")
+		}
+		var (
+			err error
+		)
+
+		namespace = fr.Namespace.Name
+
+		kvcli, err = newKubevirtClient()
+		Expect(err).ToNot(HaveOccurred())
+
+		bandwidthPerMigration := resource.MustParse("40Mi")
+		forcePostCopyMigrationPolicy := &kvmigrationsv1alpha1.MigrationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "force-post-copy",
+			},
+			Spec: kvmigrationsv1alpha1.MigrationPolicySpec{
+				AllowPostCopy:           pointer.Bool(true),
+				CompletionTimeoutPerGiB: pointer.Int64(1),
+				BandwidthPerMigration:   &bandwidthPerMigration,
+				Selectors: &kvmigrationsv1alpha1.Selectors{
+					VirtualMachineInstanceSelector: kvmigrationsv1alpha1.LabelSelector{
+						"test-live-migration": "post-copy",
+					},
+				},
+			},
+		}
+		if td.mode == kubevirtv1.MigrationPostCopy {
+			_, err = kvcli.MigrationPolicy().Create(context.TODO(), forcePostCopyMigrationPolicy, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			defer func() {
+				Expect(kvcli.MigrationPolicy().Delete(context.TODO(), forcePostCopyMigrationPolicy.Name, metav1.DeleteOptions{})).To(Succeed())
+			}()
+		}
+
+		By("Creating a test pod on both selected worker nodes")
+		podWorker1 = composeAgnhostPod(
+			"testpod-ovn-worker",
+			namespace,
+			"ovn-worker",
+			"/bin/bash", "-c", "/agnhost netexec --http-port 8000")
+		podWorker2 = composeAgnhostPod(
+			"testpod-ovn-worker2",
+			namespace,
+			"ovn-worker2",
+			"/bin/bash", "-c", "/agnhost netexec --http-port 8000")
```

Comment:
IMO, these names are misleading. 

You just want to create an HTTP server running on each of the Kubernetes nodes, right  ?

I would name this httpServerWorkerNode1 / httpServerWorkerNode2